packages/owl-bot/src/create-pr.ts

```diff
@@ -53,6 +58,56 @@ export function resplit(
   return {title, body: body.substring(0, MAX_BODY_LENGTH)};
 }
 
+const NESTED_COMMIT_SEPARATOR = 'BEGIN_NESTED_COMMIT';
+
+/**
+ * Given pull request content and a new commit message. Rewrite the pull request
+ * title and body. If the initial pull request title was truncated with ellipses,
+ * rejoin the title to the remaining part.
+ * @param {string} newCommitMessage the new commit message
+ * @param {PullRequestContent} existingContent exisiting pull request title and body
+ * @param {WithRegenerateCheckbox} withRegenerateCheckbox whether to include the
+ *   checkbox to regenerate the pull request
+ */
+export function prependCommitMessage(
+  newCommitMessage: string,
+  existingContent: PullRequestContent,
+  withRegenerateCheckbox: WithRegenerateCheckbox
+): PullRequestContent {
+  // remove any regenerate checkbox content and leading/trailing whitespace
+  const oldStrippedBody = existingContent.body
+    .replace(EMPTY_REGENERATE_CHECKBOX_TEXT, '')
+    .replace(REGENERATE_CHECKBOX_TEXT, '')
+    .trim();
+  // if title was truncated, re-add it to the beginning of the commit message
+  const oldBody = existingContent.title.endsWith('...')
+    ? `${existingContent.title.substring(
+        0,
+        existingContent.title.length - 3
+      )}${oldStrippedBody}`
+    : `${existingContent.title}\n${oldStrippedBody}`;
+  // anything before the first BEGIN_NESTED_COMMIT marker, is considered part of
+  // the previous commit
+  const bodyParts = oldBody.split(NESTED_COMMIT_SEPARATOR);
+  const oldBodyWithNestedCommitMarkers =
+    bodyParts.length === 1
+      ? // there is a single commit -- wrap the old body in the nested commit tags
+        `${NESTED_COMMIT_SEPARATOR}\n${oldBody}\nEND_NESTED_COMMIT\n`
+      : // there are already existing nested commit tags, content before the first
+        // one is wrapped in a new nested commit tag
+        `${NESTED_COMMIT_SEPARATOR}\n${
+          bodyParts[0]
+        }\nEND_NESTED_COMMIT\n${bodyParts
+          .slice(1)
+          .join(NESTED_COMMIT_SEPARATOR)}`;
+
+  // prepend the new commit message and use original title truncation logic
+  return resplit(
```

Comment:
Why not call prependCommitMessage before calling resplit?  Then resplit gets called only once and the unsplitting logic in this function can be removed.

Reply:
> We do want to rejoin that previous message to be nicer for the release notes.

Indeed, so why not call this function on the original message (before resplit)?